### PR TITLE
MBS-8944 Change TestAssertion to interface

### PR DESCRIPTION
**Pull request by sergioboi**
I will replace all inner steps in Avito repo with `action` function

### CODE REVIEW

File: subprojects/android-test/test-report/src/test/kotlin/com/avito/android/test/report/StepActionTest.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import com.avito.report.model.Entry
+import com.avito.truth.assertThat
+import com.google.common.truth.Truth.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+
+
+class StepActionTest {
```
Comment (sergioboi) [inline subprojects/android-test/test-report/src/test/kotlin/com/avito/android/test/report/StepActionTest.kt]: @eugene-krivobokov look there for scenarios

File: subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/Step.kt [hunk 2]
```diff
@@ -65,44 +71,66 @@ inline fun <T> internalStep(
...
+ * step("Step") {
+ *  action("Prepare state") {}
+ *  action("Prepare state one more time") {}
+ * }
+ *
+ * Can be called in the step scope or out of the step scope
+ * If function is called in the step scope the action will be added to the current step
+ * If function is called out of the step scope the action will be added to Synthetic step
+ */
+fun action(
```
Comment (sergioboi) [inline subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/Step.kt]: I couldn't make better name

**Comment by sergioboi** [comment]
> I will replace all inner steps in Avito repo with `action` function

We have discussed that `action` isn't needed now. We could remove internal steps

File: subprojects/common/truth-extensions/src/main/kotlin/com/avito/truth/TruthExtensions.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.avito.truth
+
+import com.google.common.truth.Truth
+
+inline fun <reified T> assertThat(any: Any, assert: T.() -> Unit) {
+    Truth.assertThat(any).isInstanceOf(T::class.java)
+    assert(any as T)
```
Comment (eugene-krivobokov) [inline subprojects/common/truth-extensions/src/main/kotlin/com/avito/truth/TruthExtensions.kt:7]: why do we need this assert?

File: subprojects/android-test/test-report/src/test/kotlin/com/avito/android/test/report/ReportSyntheticStepsTest.kt [hunk 4]
```diff
@@ -20,12 +22,15 @@ class ReportSyntheticStepsTest {
...
-    fun `when add Entries after steps than synthetic step will be created`() {
+    @BeforeEach
+    fun before() {
         // given
         report.initTestCaseHelper()
         report.startTestCase()
+    }
 
+    @Test
+    fun `when add Entries after steps than synthetic step will be created`() {
```
Comment (eugene-krivobokov) [inline subprojects/android-test/test-report/src/test/kotlin/com/avito/android/test/report/ReportSyntheticStepsTest.kt:33]: [out of scope]

We need to choose one consistent naming strategy for tests.
Examples: https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea
I see different strategies in tests. One scheme is easier to understand.

Before we were using `<state under test> <expected behaviour> <preconditions>`

`add entry after step - creates synthetic step`

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**